src/test/regress/sql/with_clause.sql

```diff
@@ -358,3 +358,63 @@ WITH RECURSIVE r1 AS (
 	)
 )
 SELECT * FROM r1 LIMIT 1;
+--
+-- Test various SELECT statements over DML operations
+--
+create table with_dml(i integer, j integer) distributed by(i);
+create table with_dml_dr(i integer, j integer) distributed replicated;
+-- Test select over insert queries. Gather motion should be there, no matter
+-- what type of distribution is used. Before fix, motion was missed for
+-- distributed replicated table, which caused count(*) to return zero rows.
+explain (costs off)
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+explain (costs off)
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+-- Test select over update queries. Check no qual pushed down to update from
+-- select, because it's incorrect to push it down in such way. Queries should
+-- be correctly executed without any rewriting error, e.g. rewriteManip.c:1409.
```

Comment:
Probably it is better to specify log message as code at [rewriteManip.c:1409](https://github.com/arenadata/gpdb/blob/2aea19802afbac1d0fa2f4b7c7d7eec5de9e3eaa/src/backend/rewrite/rewriteManip.c#L1409) can change its number later.

> "could not find replacement targetlist entry for attno %d"